sdgym/datasets.py

```diff
@@ -62,22 +63,24 @@ def _get_dataset_path(modality, dataset, datasets_path, bucket=None, aws_key=Non
     return dataset_path
 
 
-def _apply_max_columns_to_metadata(metadata, max_columns):
-    tables = metadata['tables']
-    for table in tables.values():
-        fields = table['fields']
-        if len(fields) > max_columns:
-            fields = dict(itertools.islice(fields.items(), max_columns))
-            table['fields'] = fields
+def _get_dataset_subset(data, metadata_dict):
+    if 'tables' in metadata_dict.keys():
+        raise ValueError('max_columns is not supported for multi-table datasets')
+
+    max_rows, max_columns = (1000, 10)
+    columns = metadata_dict['columns']
+    if len(columns) > max_columns:
+        columns = dict(itertools.islice(columns.items(), max_columns))
+        metadata_dict['columns'] = columns
+        data = data[columns.keys()]
 
-        structure = table.get('structure')
-        if structure:
-            structure['structure'] = structure['structure'][:max_columns]
-            structure['states'] = structure['states'][:max_columns]
+    data = data.head(max_rows)
+
+    return data, metadata_dict
 
 
 def load_dataset(modality, dataset, datasets_path=None, bucket=None, aws_key=None,
-                 aws_secret=None, max_columns=None):
+                 aws_secret=None, limit_dataset_size=None):
```

Comment:
Should we set `limit_dataset_size` to `False` by default? We could also add a `validation` that checks that `limit_dataset_size` is a boolean.

Reply:
`load_dataset` is just a helper function, only `benchmark_single_table` has access to it. I think leaving it as None is better, because it makes it clear the parameter doesn't matter by default, only if you pass a value will it do something.